repos/spack_repo/builtin/packages/py_python_lsp_server/package.py

```diff
@@ -38,4 +40,13 @@ class PyPythonLspServer(PythonPackage):
         depends_on("py-python-lsp-jsonrpc@1.0.0:1")
         depends_on("py-python-lsp-jsonrpc@1.1.0:1", when="@1.8.0:")
         depends_on("py-ujson@3.0.0:")
+        depends_on("py-black", when="@1.13:")
```

Comment:
It seems like the lsp-server only depends on `black` to validate format signatures in docstrings and can also depend on ruff for the same functionality. I'd propose we add a `formatter` variant that can either be `None`, `black`, or `ruff` that sets the relevant dependency info depending on which is chosen. (Default to None.)

https://github.com/python-lsp/python-lsp-server/commit/c8e4e992a7d0aee00804433a2211a7b3356f5b6d#diff-749bbcf2f7f5bd5289d6223d2a4cb36ac92ff1960ca2461b01cbd732ebfbd308R75-R78

Reply:
Mmh I was just following what I see in the pyproject.toml. I don't see any ruff dependency in there. Is there a reason adding black here is sub-optimal? 

I only updated this because I wanted to use it with `jupyter lab`. It isn't clear to me how or where `pylsp.signature.formatter` should be set accordinly. Do you have any idea?